modules/platforms/dotnet/Apache.Ignite.Core/Impl/Client/ClientSocket.cs

```diff
@@ -170,6 +176,29 @@ internal sealed class ClientSocket : IDisposable
 
             _features = Handshake(clientConfiguration, ServerVersion);
 
+            if (_features.HasFeature(ClientBitmaskFeature.Heartbeat) &&
+                clientConfiguration.HeartbeatInterval > TimeSpan.Zero)
+            {
+                _heartbeatInterval = clientConfiguration.HeartbeatInterval;
+
+                var serverIdleTimeout = TimeSpan.FromMilliseconds(
+                    DoOutInOp(ClientOp.GetIdleTimeout, null, r => r.Reader.ReadLong()));
+
+                if (_heartbeatInterval > serverIdleTimeout)
```

Comment:
`ClientConnectorConfiguration#getIdleTimeout` can be 0 or negative, it means that the idleTimeout is disabled. We should skip the WARN for such cases. 

I don't see such check, but I'm not familiar with .Net so maybe I miss it.

Reply:
Good catch, added a test and fixed the logic.

Reply:
As for me, it is strange decision not to compute heartbeat interval based on server idle timeout.

Reply:
@ivandasch I tried that and configuration gets too complicated. Maybe you have a better idea than this:
* `HeartbeatInterval` // `-1` is disabled (default), `0` is auto, `> 0` is manual (explicit)
* `DefaultAutoHeartbeatInterval` // when `HeartbeatInterval` is auto, but `idleTimeout` is `0`.

Reply:
I suppose that HeartbetInterval should not be greater than one-third of IdleTimeout (this is approach from ZookeeperClient).
So logic is:
1. If idleTimeout is greater than 0, and heartbeat interval is not set or larger than one-third of idle timeout -- set it as one-third of idle timeout. If hearbtBeatInterval less that one-third of idleTimeout -- use heartbeatInterval as is
2. If idleTimeout is disabled, use heartbeat interval if it is provided.

Reply:
> This will break potentially desirable behavior: don't enable heartbeats and actually drop thin clients that become idle.
I don't understand why, it would be great if you give an example of breakage

> We reset the timer on each write, see [ClientSocket.cs:854](https://github.com/apache/ignite/pull/9817/files#diff-81a8959836fbee45664e37261770ee4a215e2111c0eddd3535160ef7332232ecR854).
Great, I didn't catch it, sorry

Reply:
As a user, I've set `idleTimeout` on server so that idle client connections are closed automatically and resources are not consumed. It is up to the clients to reconnect when needed.

Now, if we enable heartbeats automatically with one-third of `idleTimeout`, those idle connections will never be dropped, and server will consume resources keeping them open.

It is just a different use case. Some users want to keep client connections open forever, even when not doing anything. Other users may want to drop them.

Too many use cases and edge cases, that's why I suggest to keep it simple and let the user decide.

Reply:
> As a user, I've set idleTimeout on server so that idle client connections are closed automatically and resources are not consumed. It is up to the clients to reconnect when needed.

Quite confusing desire as for me, cannot imagine that someone would expect this behavior. Idle client is the client that doesn't send anything, possibly stale, hang or partial dead. As for me, heartbeats is the good mechanism to notify that client is not idle, just on pause or probably waiting for some tasks.

Reply:
As for me, it makes things more complicated for user. If we choose this approach, other clients should comply this decision. Let us discuss this on dev list

Reply:
@ivandasch ok, let's continue on the dev list. Please describe your proposal there.

Reply:
Agree, let's discuss on devlist and update IEP accordingly once everyone's agree with a solution.

Reply:
As agreed on the dev list:
* `EnableHeartbeats` property added.
* `HeartbeatInterval` renamed to `DefaultHeartbeatInterval`.
* Logic and xmldocs updated.